### PR TITLE
fix(client/qbit): improve FormData requests

### PR DESCRIPTION
**Pull request by fabricionaweb**
- sets min node version to v18.14.0, lowest 18 we can go to work with those apis
- improves FormData requests to qBittorrent by using Blob to send the files
- uses native fetch and formData

## Summary by CodeRabbit

- **Refactor**
  - Updated the app to be compatible with Node.js version 20.

- **Documentation**
  - Updated the README to reflect the new minimum required Node.js version.

- **Chores**
  - Removed unnecessary `node-fetch` imports to align with Node.js built-in features.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
       -
         uses: actions/setup-node@v3
         with:
-          node-version: 18
+          node-version: 18.18.2
```
Comment (onedr0p) [inline .github/workflows/ci.yml:26]:
```suggestion
          node-version: 18
```

I would suggest keeping this `18` (which is the latest 18 release) so it doesn't need to be managed unless it needs to be updated to a newer major version (e.g. 20)

File: package.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
 		"cross-seed": "dist/cmd.js"
 	},
 	"engines": {
-		"node": ">=16"
+		"node": ">=18.18.2"
```
Comment (onedr0p) [inline package.json:17]:
```suggestion
		"node": ">=18"
```

Same reasons as above, I would set this to `>=18`

File: src/cmd.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env node
+#!/usr/bin/env node --no-warnings
```
Comment (mmgoodnow) [inline src/cmd.ts:1]: smart, I didn't know this worked!

File: src/clients/QBittorrent.ts [hunk 4]
```diff
@@ -117,9 +112,9 @@ export default class QBittorrent implements TorrentClient {
 			);
 		}
 
-		const cookieArray = response.headers.raw()["set-cookie"];
+		const cookieArray = response.headers.get("set-cookie");
```
Comment (mmgoodnow) [inline src/clients/QBittorrent.ts:115]: small style nit - now that we're accessing a `Headers` object with `.get`, I think we can use the proper casing (previously everything was lowercased in order to support case insensitivity)

```suggestion
		const cookieArray = response.headers.get("Set-Cookie");
```

File: README.md [hunk 5]
```diff
@@ -19,7 +19,7 @@ delivery into your client.
 
 ## Requirements
 
--   [Node 16+](https://nodejs.org/en/download)
+-   [Node >= 18.14.0](https://nodejs.org/en/download)
```
Comment (mmgoodnow) [inline README.md:22]:
```suggestion
-   [Node >= 20](https://nodejs.org/en/download)
```

File: .nvmrc [hunk 6]
```diff
@@ -1 +1 @@
-v16
+v18.14.0
```
Comment (mmgoodnow) [inline .nvmrc:1]:
```suggestion
v20
```

File: package.json [hunk 7]
```diff
@@ -14,7 +14,7 @@
 		"cross-seed": "dist/cmd.js"
 	},
 	"engines": {
-		"node": ">=16"
+		"node": ">=18.14.0"
```
Comment (mmgoodnow) [inline package.json:17]:
```suggestion
		"node": ">=20"
```

**Review by mmgoodnow — COMMENTED** [review]
LGTM pending these nits. I changed the base branch to `v6-unstable` which will act as a staging area for a batch of PRs with breaking changes

File: src/clients/QBittorrent.ts (hunk 4 above)
Comment (mmgoodnow) [reply src/clients/QBittorrent.ts]: ah good catch. after reading up it seems like native fetch in node will not expose Set-Cookie headers to us directly in order to preserve compatibility with browser `fetch`.

**Review by mmgoodnow — APPROVED** [review]

**Pull request merged**